haas/test_common.py

```diff
@@ -14,12 +14,80 @@
 
 from functools import wraps
 from haas.model import *
-# XXX: This function has an underscore so that we don't import it elsewhere.
-# But... we need it here.  Oops.
-from haas.model import _on_virt_uri
 from haas.config import cfg
-from haas import api
+from haas import api, config
 import json
+import subprocess
+import os.path
+
+
+def testsuite_config():
+    """A pytest decorator which loads an initial config from ``testsuite.cfg``.
+
+    Tests which don't care about a specific configuration should leave the
+    config alone. This allows the developer to test with different
+    configurations, e.g. different DBMS backends.
+
+    if testsuite.cfg doesn't exist, sane defaults are provided.
+    """
+    # NOTE: The file ``testsuite.cfg.default`` Should be updated whenever
+    # The default settings here are modified.
+    if os.path.isfile('testsuite.cfg'):
+        config.load('testsuite.cfg')
+    else:
+        config_set({
+            'general': {
+                'driver': 'null',
+            },
+            'devel': {
+                'dry_run': True,
+            },
+            'headnode': {
+                'base_imgs': 'base-headnode, img1, img2, img3, img4',
+            },
+            'database': {
+                'uri': 'sqlite:///:memory:',
+            },
+        })
+
+
+def config_merge(config_dict):
+    """Modify the configuration according to ``config_dict``.
+
+    ``config_dict`` should be a dictionary mapping section names (strings)
+    to dictionaries mapping option names within a section (again, strings)
+    to their values. If the value of a section, or option is None, that
+    section or option is removed. Otherwise, the section is created if it
+    does not exist, and any options are set to the specified values.
+    """
+    for section in config_dict.keys():
+        if config_dict[section] is None:
+            cfg.remove_section(section)
```

Comment:
It's odd that they throw an exception for creating an existing section but not one for deleting a non-existent one.  Okay, though